src/widgets/DataTable/DataTableRow.js

```diff
@@ -102,22 +102,45 @@ const DataTableRow = React.memo(
           switch (type) {
             case COLUMN_TYPES.EDITABLE_STRING:
             case COLUMN_TYPES.EDITABLE_NUMERIC: {
-              // Special condition for stocktake counted total quantity cells.
-              // Use the placeholder 'Not counted' when a stocktake item or batch
-              // has not been counted yet.
-              let placeholder = '';
-              if (columnKey === COLUMN_KEYS.COUNTED_TOTAL_QUANTITY) {
-                placeholder = rowData.hasBeenCounted ? '' : tableStrings.not_counted;
-              }
+              const { isVaccine, hasBeenCounted } = rowData;
+
+              // Placeholder values which are determined on a per row basis. For example, display
+              // "Not counted" as a placeholder on a stocktake row. Placeholders are used under
+              // the condition [placeholder && !value] which can be a source of bugs with JS
+              // falsey values. I.e. if a value is 0, the placeholder will display rather than
+              // the value.
+              const extraPlaceholders = {
+                [COLUMN_KEYS.COUNTED_TOTAL_QUANTITY]: hasBeenCounted
+                  ? ''
+                  : tableStrings.not_counted,
+              };
+
+              // Extra disabled conditions for columns which can be determined per row. For example,
+              // only allow the doses column for vaccine items.
+              const disabledCondition = {
+                [COLUMN_KEYS.DOSES]: !isVaccine,
+              };
+
+              // Extra text which can be displayed for a column on a per row basis when the
+              // cell is disabled. For example display N/A on a row for the doses column rather
+              // than the underlying value "0".
+              const disabledText = {
+                [COLUMN_KEYS.DOSES]: isVaccine ? '' : generalStrings.not_available,
+              };
+
+              const value = disabledText[columnKey] ? disabledText[columnKey] : rowData[columnKey];
+              const placeholder = extraPlaceholders[columnKey] ?? '';
+
+              const inputIsDisabled = isDisabled || !!disabledCondition[columnKey];
```

Comment:
A little weird and not very nice to be in this component, really. But here it is, open to suggestions. Basically want to add a row-level disable condition, different text when disabled (to display N/A on a doses column rather than the underlying value of 0). Thought this was a niceish implementation for extending?

Reply:
agree, not ideal to have in the component. I wondered about passing in functions to do the placeholder, disabled condition and disabled text - ones which accept the `rowData` as a parameter. They could be passed in as props?

Reply:
is fine as it is, just wondering about alternatives to keep some of this specificity out of the more general DataTable code